youtube_dl/extractor/npo.py

```diff
@@ -82,686 +27,253 @@ class NPOIE(NPOBaseIE):
-        },
-    }
+    _TESTS = [{
+        'url': 'https://schooltv.nl/item/zapp-music-challenge-2015-zapp-music-challenge-2015',
+        # TODO fill in other test attributes
+    }]
 
     def _real_extract(self, url):
-        audio_id = self._match_id(url)
+        video_id = url.rstrip('/').split('/')[-1]
 
-        webpage = self._download_webpage(url, audio_id)
+        # TODO Find out how we could obtain this automatically
+        #      Otherwise this extractor might break each time SchoolTV deploys a new release
+        build_id = 'b7eHUzAVO7wHXCopYxQhV'
 
-        title = self._html_search_regex(
-            r'href="/radio/[^/]+/fragment/%s" title="([^"]+)"' % audio_id,
-            webpage, 'title')
+        metadata_url = 'https://schooltv.nl/_next/data/' \
+                       + build_id \
+                       + '/item/' \
+                       + video_id + '.json'
 
-        audio_url = self._search_regex(
-            r"data-streams='([^']+)'", webpage, 'audio url')
+        metadata = self._download_json(metadata_url,
+                                       video_id).get('pageProps', {}).get('data', {})
 
-        return {
-            'id': audio_id,
-            'url': audio_url,
-            'title': title,
-        }
+        formats = self._download_by_product_id(metadata.get('poms_mid'), video_id)
 
+        if not formats:
+            raise ExtractorError('Could not find a POMS product id in the provided URL, '
+                                 'perhaps because all stream URLs are DRM protected.')
 
-class NPODataMidEmbedIE(InfoExtractor):
-    def _real_extract(self, url):
-        display_id = self._match_id(url)
-        webpage = self._download_webpage(url, display_id)
-        video_id = self._search_regex(
-            r'data-mid=(["\'])(?P<id>(?:(?!\1).)+)\1', webpage, 'video_id', group='id')
         return {
-            '_type': 'url_transparent',
-            'ie_key': 'NPO',
-            'url': 'npo:%s' % video_id,
-            'display_id': display_id
-        }
-
-
-class SchoolTVIE(NPODataMidEmbedIE):
-    IE_NAME = 'schooltv'
-    _VALID_URL = r'https?://(?:www\.)?schooltv\.nl/video/(?P<id>[^/?#&]+)'
-
-    _TEST = {
-        'url': 'http://www.schooltv.nl/video/ademhaling-de-hele-dag-haal-je-adem-maar-wat-gebeurt-er-dan-eigenlijk-in-je-lichaam/',
-        'info_dict': {
-            'id': 'WO_NTR_429477',
-            'display_id': 'ademhaling-de-hele-dag-haal-je-adem-maar-wat-gebeurt-er-dan-eigenlijk-in-je-lichaam',
-            'title': 'Ademhaling: De hele dag haal je adem. Maar wat gebeurt er dan eigenlijk in je lichaam?',
-            'ext': 'mp4',
-            'description': 'md5:abfa0ff690adb73fd0297fd033aaa631'
-        },
-        'params': {
-            # Skip because of m3u8 download
-            'skip_download': True
+            'id': video_id,
+            'title': metadata.get('title', '') + ' - ' + metadata.get('subtitle', ''),
```

Comment:
Or use `utils.join_nonempty()` for this effect:
* `'title'`, `'subtitle'` -> `'title - subtitle'`
* `'title'`, `''` -> `'title'`
* `''`, `'subtitle'` -> `'subtitle'`

Reply:
Done!